programs/bench.c

```diff
@@ -182,32 +183,34 @@ typedef struct {
     size_t resSize;
 } blockParam_t;
 
-
-
 #undef MIN
 #undef MAX
 #define MIN(a,b)    ((a) < (b) ? (a) : (b))
 #define MAX(a,b)    ((a) > (b) ? (a) : (b))
 
-static int BMK_benchMem(const void* srcBuffer, size_t srcSize,
+BMK_result_t BMK_benchMem(const void* srcBuffer, size_t srcSize,
```

Comment:
minor : 
I believe it's worthwhile to re-order arguments in a way which makes a bit more sense.
It's becoming a complex function, with an uncomfortably large nb of arguments, 
and more importantly, it's becoming a public (exposed) function, 
so it matters.
Here is my suggestion : 
````
const void* srcBuffer, size_t srcSize,        // like now
const size_t* fileSizes, unsigned nbFiles,    // describes the content of `srcBuffer`, so related. Also : prefer `unsigned` over `U32`, since the prototype is declared in `bench.h`
int cLevel, const ZSTD_compressionParameters* const comprParams,// they are related, keep them together
const void* dictBuffer, size_t dictBufferSize,// complementary compression argument
ZSTD_CCtx* ctx, ZSTD_DCtx* dctx,              // workspace contexts
int displayLevel, const char* displayName     // both are about display, they are related
````